batcher.js

```diff
@@ -13,22 +13,22 @@ if (process.argv.length < 4) {
 
 var command = process.argv[2];
 var query = process.argv[3];
-var data = parseData(process.argv[4]);
+var searchParams = sToO(process.argv[4]);
+var data = sToO(process.argv[5]);
 
-function parseData(data) {
-    // FIXME: not particularly typesafe.
-    try {
-        return JSON.parse(data)
-    } catch(e) {
-        return undefined;
+function sToO(s) {
+    // Wow!
+    eval('const o = '+ s);
+    if (typeof o === 'object') {
+        return o;
     }
 }
 
 System.import('./core').then(function(core) {
-    core.execute(query, command, data).then(function() {
+    core.execute(command, query, searchParams, data).then(function() {
         process.exit(0);
     }).catch(function(error) {
         console.error(error.stack);
         process.exit(1);
     });
-});
+}).catch(function(e) { throw e; });
```

Comment:
What is this meant to do? AFAICS it will just make a rejected promise into a rejected promise..?